### PR TITLE
[AMORO-2018][Flink] Apply spotless plugin to Flink modules

### LINKED ISSUE #2018: [Subtask]: Apply spotless plugin to Flink modules
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### Description

Apply spotless plugin to Flink modules and set `mvn validate` as spotless:check, remove checkstyle plugin.

### Parent issue

#1402

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhongqishang**
## Why are the changes needed?

Close #2018 .

## Brief change log

- Apply spotless plugin to Flink modules and set mvn validate as spotless:check (default)
- Skip checkstyle plugin.
- Via `mvn spotless:apply -pl flink -amd` to format flink module.

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / **no**)
- If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Review by baiyangtx — APPROVED** [review]
LGTM, thanks for your contribution

File: flink/v1.12/flink/src/main/java/com/netease/arctic/flink/read/LogReadHelper.java [hunk 1]
```diff
@@ -110,27 +110,21 @@ public void initializeState(FunctionInitializationContext context, RuntimeContex
...
   /**
-   * turn row kind of a row.
-   * +I -> -D
-   * -D -> +I
-   * -U -> +U
-   * +U -> -U
+   * turn row kind of a row. +I -> -D -D -> +I -U -> +U +U -> -U
    *
    * @param rowData before reset row
    * @return after reset row kind.
```
Comment (baiyangtx) [inline flink/v1.12/flink/src/main/java/com/netease/arctic/flink/read/LogReadHelper.java:130]: I think those formats should be kept.

File: flink/v1.12/flink/src/main/java/com/netease/arctic/flink/read/hybrid/assigner/ShuffleSplitAssigner.java [hunk 2]
```diff
@@ -226,17 +237,12 @@ void deserializePartitionIndex(long[] shuffleSplitRelation) {
...
-   * Different data files may locate in different layers when multi snapshots are committed, so arctic source reading
-   * should consider emitting the records and keeping ordering. According to the dataTreeNode of the arctic split and
-   * the currentMaskOfTreeNode, return the exact tree node list which may move up or go down layers in the arctic tree.
-   * </p>
+   * |mask=0 o | / \ |mask=1 o o | / \ / \ |mask=3 o o o o
+   *
+   * <p>Different data files may locate in different layers when multi snapshots are committed, so
+   * arctic source reading should consider emitting the records and keeping ordering. According to
+   * the dataTreeNode of the arctic split and the currentMaskOfTreeNode, return the exact tree node
+   * list which may move up or go down layers in the arctic tree.
```
Comment (baiyangtx) [inline flink/v1.12/flink/src/main/java/com/netease/arctic/flink/read/hybrid/assigner/ShuffleSplitAssigner.java:245]: I think those formats should be kept.

Comment (zhongqishang) [reply flink/v1.12/flink/src/main/java/com/netease/arctic/flink/read/hybrid/assigner/ShuffleSplitAssigner.java]: 1

File: flink/v1.12/flink/src/main/java/com/netease/arctic/flink/read/LogReadHelper.java (hunk 1 above)
Comment (zhongqishang) [reply flink/v1.12/flink/src/main/java/com/netease/arctic/flink/read/LogReadHelper.java]: 1

**Review by baiyangtx — APPROVED** [review]
LGTM

**Pull request merged**